File: Backend/Core/Business/Backend.Business-ExerciseType/ExerciseTypeRequests/Create/CreateExerciseTypeRequest.cs
using Backend.Domain.Entities.Exercises;
using MediatR;
using System;
using System.Collections.Generic;

namespace Backend.Business.Exercises.ExerciseTypeRequests.Create
{
    public class CreateExerciseTypeRequest : IRequest<ExerciseType>
    {
        public string Name { get; set; }

        public bool Active { get; set; }

        public bool? RequiresReps { get; set; }
        public bool? RequiresSets { get; set; }
        public bool? RequiresWeight { get; set; }
        public bool? RequiresBodyweight { get; set; }
        public bool? RequiresTime { get; set; }

        public Guid ApplicationUserId { get; set; }

        public IEnumerable<ExerciseTypeTag> Properties { get; set; }
    }
}

File: Frontend/src/server-models/cqrs/exercise-type/create-exercise-type.request.ts
import { ExerciseTypeTag } from 'src/server-models/entities/exercise-type.model';
export class CreateExerciseTypeRequest {

    constructor(data: Partial<CreateExerciseTypeRequest>) {
        Object.assign(this, data);
    }

    name: string;
    active: boolean;
    requiresReps?: boolean;
    requiresWeight?: boolean;
    requiresSets?: boolean;
    requiresBodyweight?: boolean;
    requiresTime?: boolean;
    properties: ExerciseTypeTag[];
    applicationUserId: string;
}